restaurant-menu/src/containers/CustomerInfo/CustomerInfo.js

```diff
@@ -0,0 +1,360 @@
+                                        onChange={event => onChangeHandler(event, "street")}
+                                    />
+                                </div>
+                            </div>
+                            <div className="seperator"></div>
+                            <div className="input-couple">
+                                <div className={submitValidation.number ? "info" : "info invalid"}>
+                                    <label>* מספר בית</label>
+                                    <input
+                                        type="number"
+                                        min="1" 
+                                        placeholder="מספר בית"
+                                        step="1"
+                                        value={order.number}
+                                        onChange={event => onChangeHandler(event, "number")}
+                                    />
+                                </div>
+                                <div className="seperator"></div>
+                                <div className="info">
+                                    <label>מספר דירה</label>
+                                    <input
+                                        type="number"
+                                        min="1"
+                                        placeholder="מספר דירה"
+                                        value={order.apartment}
+                                        onChange={event => onChangeHandler(event, "apartment")}
+                                    />
+                                </div>
+                            </div>
+                        </div>
+
+                        <h3>הערות להזמנה</h3>
+                        <div className="remark-section">
+                            <input
+                                type="text"
+                                placeholder="הערות"
+                                value={order.remark}
+                                onChange={event => onChangeHandler(event, "remark")}
+                            />
+                        </div>
+                        <div className="submit-section">
+                            <button type="submit">בצע הזמנה</button>
+                        </div>
+                        <div className="invalidMessage">
+                            {invalidMessage}     
+                        </div>
+                    </form>
+                </div>
+            </React.Fragment>
+        );
+    }
+    else if(status === "loading")
+        content = (
+            <div className="spinner-and-error">
+                <Spinner/>
+            </div>
+        );
+    else if(status === "error")
+        content = (
+            <div className="spinner-and-error">
+                <p>שגיאת תקשורת, נא לנסות מאוחר יותר</p>
+            </div>
+        );
+
+    return content;
+}
+
+export default CustomerInfo;
```

Comment:
Unrelated to the above, make sure you know what is useEffect, useMemo, useCallback, how are they used to make renders more efficient. How does the immutability principle help those hooks? What is React.memo? When should you place functions inside your component and when should you place them outside, and what is the disadvantage of having them inside? What happens if you place a function inside and pass it as a prop to another component? What is a "pure" component?